src/python/pants/goal/run_tracker.py

```diff
@@ -503,3 +518,20 @@ def _get_option_to_record(self, scope, option=None):
             raise ValueError(
                 f"Couldn't find option scope {scope}{option_str} for recording ({e!r})"
             )
+
+    def retrieve_logs(self) -> List[str]:
+        """Get a list of every log entry recorded during this run."""
+
+        if not self.run_logs_file:
+            return []
+
+        output = []
+        try:
+            with open(self.run_logs_file, "r") as f:
+                for line in f:
+                    output.append(line)
+        except OSError as e:
+            logger.warning("Error retrieving per-run logs from RunTracker:")
```

Comment:
you can just do `exc_info=true` here instead of adding two logs statements.
https://docs.python.org/3/library/logging.html#logging.debug